examples/typescript/package.json

```diff
@@ -5,14 +5,14 @@
   "dependencies": {
     "@types/jest": "^26.0.15",
     "@types/node": "^11.15.35",
-    "@types/react": "^16.9.55",
+    "@types/react": "^17.0.0",
     "@types/react-dom": "^16.9.9",
     "graphql-hooks": "^5.4.1",
     "graphql-hooks-memcache": "^2.2.1",
-    "react": "^17.0.1",
-    "react-dom": "^17.0.1",
-    "react-scripts": "^3.4.4",
-    "typescript": "^4.0.5"
+    "react": "^17.0.0",
+    "react-dom": "^17.0.0",
+    "react-scripts": "^4.0.3",
```

Comment:
Upgrading react-scripts allows an higher peer dependency of TypeScript.
Both the examples with create-react-app are compiling and working.